### PR TITLE
Added several tests for SciPy

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Fixes: https://github.com/lfortran/lfortran/issues/1795
Fixes: https://github.com/lfortran/lfortran/issues/162
Fixes: https://github.com/lfortran/lfortran/issues/1090
Fixes: https://github.com/lfortran/lfortran/issues/1688
Works already with latest main - 
fixed: https://github.com/lfortran/lfortran/issues/1530
fixed: https://github.com/lfortran/lfortran/issues/370
Fixes: #2177

### CODE REVIEW

File: integration_tests/procedure_05.f90 [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FUNCTION DNAN()
```
Comment (Pranavchiku) [inline integration_tests/procedure_05.f90:1]: can you please convert it to lower case?

File: integration_tests/procedure_05.f90 [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    Z = DCMPLX(X, Y)
+
+    XC=ABS(Z)
+    LS=0
+    IF (DIMAG(Z).EQ.0.0D0.OR.XC.LT.1.0D0) LS=1
+
+    RETURN
+END
+program main
+end program main
```
Comment (Pranavchiku) [inline integration_tests/procedure_05.f90:18]: Let's call subroutine `clqmn` here and test `ls` like `if (LS /= <output of gfortran> ) error stop`

File: integration_tests/procedure_05.f90 [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+SUBROUTINE CLQMN(X,Y)
+    IMPLICIT DOUBLE PRECISION (X,Y)
+    IMPLICIT COMPLEX*16 (C,Z)
+    Z = DCMPLX(X, Y)
+
+    XC=ABS(Z)
+    LS=0
+    IF (DIMAG(Z).EQ.0.0D0.OR.XC.LT.1.0D0) LS=1
+
```
Comment (Pranavchiku) [inline integration_tests/procedure_05.f90:14]: test `ls` like `if (LS /= <output of gfortran> ) error stop`

File: integration_tests/save_04.f90 [hunk 4]
```diff
@@ -0,0 +1,7 @@
+subroutine idd_random_transf_init()
+    implicit real *8 (a-h,o-z)
+    save
+    return
+end
+program main
+end program main
```
Comment (Pranavchiku) [inline integration_tests/save_04.f90:7]: Can we make the test more strong, by assigning some value and checking it in the subroutine?

**Comment by Pranavchiku** [comment]
Converting this PR to draft.

File: integration_tests/min_three_args.f90 [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+        tmp2 = huge(1.d0)  
+        arr(:,1) = min( 0.d0,tmp1,tmp2 )   
+
+        if (arr(1, 1) /= 0.d0) error stop
+        if (size(arr) /= 2) error stop
+        
+    end subroutine take_min
+end module mintest
+program name
+end program name
```
Comment (Pranavchiku) [inline integration_tests/min_three_args.f90:17]:
```suggestion
program name
! Use module mintest and call subroutine `take_min`
end program name
```
If feel you can call the subroutine `take_min`.

**Review by Pranavchiku — APPROVED** [review]
I left a comment, rest looks good to me, thanks!

File: integration_tests/min_three_args.f90 (hunk 5 above)
Comment (HarshitaKalani) [reply integration_tests/min_three_args.f90]: Calling subroutine 'take_min' in the main program throws compiler error. I've opened a new issue #2177 for that.

File: integration_tests/min_three_args.f90 [hunk 6]
```diff
@@ -0,0 +1,21 @@
+module mintest
```
Comment (certik) [inline integration_tests/min_three_args.f90:1]:
```suggestion
module min_three_args
```

**Comment by certik** [comment]
This makes it work on macOS:
```diff
--- a/integration_tests/min_three_args.f90
+++ b/integration_tests/min_three_args.f90
@@ -3,10 +3,11 @@ module min_three_args
 contains 
     subroutine take_min(arr)   
         real(8), intent(out) :: arr(:,:)  
-        real(8) :: tmp1, tmp2  
+        real(8) :: tmp1, tmp2, tmp3  
         tmp1 = huge(1.d0)
         tmp2 = huge(1.d0)
-        arr(:,1) = min( 0.d0,tmp1,tmp2 )   
+        tmp3 = min( 0.d0,tmp1,tmp2 )   
+        arr(:,1) = tmp3
 
         if (arr(1, 1) /= 0.d0) error stop
         if (size(arr) /= 2) error stop
```
This is a bug in assigning min to an array, that we need to fix.

**Review by czgdp1807 — APPROVED** [review]

**Comment by certik** [comment]
Thanks!

**Pull request merged**